### ISSUE TITLE
Painted Turtles are flagged as dangerous

### ISSUE THREAD

**Issue by failboat78** [CleverRaven/Cataclysm-DDA#76409]
### Describe the bug

Seeing a lot of these new critters - they interrupt travel with danger warnings. Is this intended?

I am no biologist but they don't seem very dangerous to me? https://en.wikipedia.org/wiki/Painted_turtle

### Attach save file

N/A

### Steps to reproduce

Wander around - they are very common

### Expected behavior

Harmless critters should not trigger warnings

### Screenshots

![image](https://github.com/user-attachments/assets/ce960643-81e8-48d9-a3fe-46ea82fbeee3)

### Versions and configuration

- OS: Linux
    - OS Version: <unknown>
- Game Version: cdda-experimental-2024-09-12-1816 2473620 [64-bit]
- Graphics Version: Tiles
- Game Language: English [en]
- Mods loaded: [
    Dark Days Ahead [dda],
    Disable NPC Needs [no_npc_food],
    Portal Storms Ignore NPCs [personal_portal_storms],
    Slowdown Fungal Growth [no_fungal_growth],
    Mythical Martial Arts [MMA],
    Bombastic Perks [bombastic_perks],
    Mind Over Matter [mindovermatter]
]

**Comment by Karol1223** [comment]
It's a known issue, not exclusive to turtles. Safemode manager has a lower aggression threshold than actual aggression threshold is ingame. Monsters become aggressive at ``aggression`` being above 10, safemode triggers when ``aggression`` is above 0. There's a bunch of monsters affected by this.

**Comment by RenechCDDA** [comment]
> It's a known issue, not exclusive to turtles. Safemode manager has a lower aggression threshold than actual aggression threshold is ingame. Monsters become aggressive at `aggression` being above 10, safemode triggers when `aggression` is above 0. There's a bunch of monsters affected by this.
> 
> Note that snapping turtles are also flagged as dangerous but that _is_ intentional, it's the painted ones that are a victim of safemode being wrong. You'll find the same is true for several new snake species, and until recently rats.

Possible patch:

```diff
index ccfafa25b6..43ae018c11 100644
--- a/src/game.cpp
+++ b/src/game.cpp
@@ -4456,7 +4456,8 @@ void game::mon_info_update( )
                                       mon_dist,
                                       u.controlling_vehicle ) == rule_state::BLACKLISTED;
             } else {
-                need_processing =  MATT_ATTACK == matt || MATT_FOLLOW == matt;
+                need_processing =  MATT_ATTACK == matt ||
+                                   ( MATT_FOLLOW == matt && critter.get_dest() == u.get_location() );
             }
             if( need_processing ) {
                 if( index < 8 && critter.sees( get_player_character() ) ) {
```

**Comment by Psithief** [comment]
> It's a known issue, not exclusive to turtles. Safemode manager has a lower aggression threshold than actual aggression threshold is ingame. Monsters become aggressive at `aggression` being above 10, safemode triggers when `aggression` is above 0. There's a bunch of monsters affected by this.
> 
> Note that snapping turtles are also flagged as dangerous but that _is_ intentional, it's the painted ones that are a victim of safemode being wrong. You'll find the same is true for several new snake species, and until recently rats.

Snapping turtles are only ever angry because people pick them up incorrectly. They should not be more aggressive than regular animals unless it's a mutated version. Ref:
[We Have Been Completely Wrong About Snapping Turtles!](https://www.youtube.com/watch?v=CqPvKx86LuI) YouTube - Clint's Reptiles.

**Comment by Karol1223** [comment]
> Snapping turtles are only ever angry because people pick them up incorrectly. They should not be more aggressive than regular animals

Yeah nah. If we keep rattlesnakes being aggressive when a player walks close, so will snapping turtles. In an ideal world we'd have a trigger that only makes them potentially attack when you're on an adjacent tile, and never chase you, but that's not possible currently.
